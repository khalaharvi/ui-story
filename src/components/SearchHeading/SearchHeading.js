import React from 'react';
import PropTypes from 'prop-types';
import { connectStateResults } from 'react-instantsearch/connectors';

import {
  PRIMARY_CATEGORY_KEY,
  SECONDARY_CATEGORY_KEY,
  TERTIARY_CATEGORY_KEY
} from 'utils/constants';

const SEARCH_STATE_KEY = 'hierarchicalMenu';

export const SearchHeading = ({ defaultHeading, searchState }) => {
  let heading = '';

  if (searchState &&
      searchState[SEARCH_STATE_KEY] &&
      searchState[SEARCH_STATE_KEY][PRIMARY_CATEGORY_KEY] !== ''
  ) {
    const menu = searchState[SEARCH_STATE_KEY];

    if (menu[TERTIARY_CATEGORY_KEY]) {
      heading = menu[TERTIARY_CATEGORY_KEY];
    }

    if (menu[SECONDARY_CATEGORY_KEY]) {
      heading += menu[SECONDARY_CATEGORY_KEY];
    }

    if (menu[PRIMARY_CATEGORY_KEY]) {
      heading += menu[PRIMARY_CATEGORY_KEY];
    }
  } else {
    heading = defaultHeading;
  }

  heading = heading.split(' < ').pop();

  return <span>{heading}</span>;
};

SearchHeading.propTypes = {
  searchState: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  defaultHeading: PropTypes.string
};

SearchHeading.defaultProps = {
  defaultHeading: 'Search'
};

export default connectStateResults(SearchHeading);
