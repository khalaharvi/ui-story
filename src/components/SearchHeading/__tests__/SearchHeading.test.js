import React from 'react';
import { shallow } from 'enzyme';

import { SearchHeading } from '../SearchHeading';

describe('<SearchHeading />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SearchHeading searchState={{}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with defaultHeading', () => {
    const wrapper = shallow(<SearchHeading searchState={{}} defaultHeading="Furnishings" />);
    expect(wrapper).toMatchSnapshot();
  });
});
