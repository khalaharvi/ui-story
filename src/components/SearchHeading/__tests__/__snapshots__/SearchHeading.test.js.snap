// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchHeading /> renders correctly 1`] = `
<span>
  Search
</span>
`;

exports[`<SearchHeading /> renders correctly with defaultHeading 1`] = `
<span>
  Furnishings
</span>
`;
